src/complex/Utilities/FilterUtilities.cpp

```diff
@@ -23,4 +23,18 @@ Result<> CreateOutputDirectories(const fs::path& outputPath)
   }
   return {};
 }
+
+// -----------------------------------------------------------------------------
+void AppendDataModifiedActions(const DataStructure& dataStructure, std::vector<DataModifiedAction>& modifiedActions, const DataPath& parentPath, const std::vector<DataPath>& ignoredDataPaths)
+{
+  std::optional<std::vector<DataPath>> result = complex::GetAllChildArrayDataPaths(dataStructure, parentPath, ignoredDataPaths);
+  if(result.has_value())
+  {
+    for(const auto& child : result.value())
+    {
+      modifiedActions.emplace_back(DataModifiedAction{child, DataModifiedAction::ModifiedType::DataArrayModified});
```

Comment:
```suggestion
      modifiedActions.emplace_back(child, DataModifiedAction::ModifiedType::DataArrayModified);
```

Reply:
Clang does not like that syntax. Keeping it the same

Reply:
Do you have a diagnostic message?

Reply:
```
====================[ Build | all | NX-x64-Debug ]==============================
/opt/local/cmake/CMake.app/Contents/bin/cmake --build /Users/mjackson/Workspace1/DREAM3D-Build/NX-x64-Debug --target all -j 14
[7/39] Building CXX object complex/CMakeFiles/complex.dir/src/complex/Utilities/FilterUtilities.cpp.o
FAILED: complex/CMakeFiles/complex.dir/src/complex/Utilities/FilterUtilities.cpp.o 
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -DCOMPLEX_ENABLE_MULTICORE -DFMT_SHARED -DH5Support_USE_MUTEX -DH5_BUILT_AS_DYNAMIC_LIB -DH5_USE_110_API -DTBB_USE_DEBUG -Dcomplex_EXPORTS -I/Users/mjackson/Workspace1/complex/src -I/Users/mjackson/Workspace1/DREAM3D-Build/NX-x64-Debug/complex/generated -isystem /Users/mjackson/Workspace1/vcpkg-installed/x64-osx-v11/include -isystem /Users/mjackson/Workspace1/vcpkg-installed/x64-osx-v11/share/eigen3/../../include/eigen3 -g -std=c++17 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.1.sdk -mmacosx-version-min=11.0 -fPIC -fcolor-diagnostics -Werror=parentheses -Werror=return-type -Werror=shadow-all -MD -MT complex/CMakeFiles/complex.dir/src/complex/Utilities/FilterUtilities.cpp.o -MF complex/CMakeFiles/complex.dir/src/complex/Utilities/FilterUtilities.cpp.o.d -o complex/CMakeFiles/complex.dir/src/complex/Utilities/FilterUtilities.cpp.o -c /Users/mjackson/Workspace1/complex/src/complex/Utilities/FilterUtilities.cpp
In file included from /Users/mjackson/Workspace1/complex/src/complex/Utilities/FilterUtilities.cpp:1:
In file included from /Users/mjackson/Workspace1/complex/src/complex/Utilities/FilterUtilities.hpp:3:
In file included from /Users/mjackson/Workspace1/complex/src/complex/Utilities/ParallelAlgorithmUtilities.hpp:3:
In file included from /Users/mjackson/Workspace1/complex/src/complex/Common/Types.hpp:5:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.1.sdk/usr/include/c++/v1/set:439:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.1.sdk/usr/include/c++/v1/__node_handle:15:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.1.sdk/usr/include/c++/v1/memory:674:
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.1.sdk/usr/include/c++/v1/__memory/allocator.h:154:28: error: no matching constructor for initialization of 'complex::DataObjectModificationNotification'
        ::new ((void*)__p) _Up(_VSTD::forward<_Args>(__args)...);
                           ^   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.1.sdk/usr/include/c++/v1/__memory/allocator_traits.h:290:13: note: in instantiation of function template specialization 'std::allocator<complex::DataObjectModificationNotification>::construct<complex::DataObjectModificationNotification, const complex::DataPath &, complex::DataObjectModificationNotification::ModifiedType>' requested here
        __a.construct(__p, _VSTD::forward<_Args>(__args)...);
            ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.1.sdk/usr/include/c++/v1/vector:948:21: note: in instantiation of function template specialization 'std::allocator_traits<std::allocator<complex::DataObjectModificationNotification>>::construct<complex::DataObjectModificationNotification, const complex::DataPath &, complex::DataObjectModificationNotification::ModifiedType, void>' requested here
    __alloc_traits::construct(this->__alloc(), _VSTD::__to_address(__tx.__pos_),
                    ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.1.sdk/usr/include/c++/v1/vector:1706:9: note: in instantiation of function template specialization 'std::vector<complex::DataObjectModificationNotification>::__construct_one_at_end<const complex::DataPath &, complex::DataObjectModificationNotification::ModifiedType>' requested here
        __construct_one_at_end(_VSTD::forward<_Args>(__args)...);
        ^
/Users/mjackson/Workspace1/complex/src/complex/Utilities/FilterUtilities.cpp:39:21: note: in instantiation of function template specialization 'std::vector<complex::DataObjectModificationNotification>::emplace_back<const complex::DataPath &, complex::DataObjectModificationNotification::ModifiedType>' requested here
    modifiedActions.emplace_back(child, DataObjectModificationNotification::ModifiedType::DataArrayModified);
                    ^
/Users/mjackson/Workspace1/complex/src/complex/Filter/Output.hpp:113:23: note: candidate constructor (the implicit copy constructor) not viable: requires 1 argument, but 2 were provided
struct COMPLEX_EXPORT DataObjectModificationNotification
                      ^
/Users/mjackson/Workspace1/complex/src/complex/Filter/Output.hpp:113:23: note: candidate constructor (the implicit move constructor) not viable: requires 1 argument, but 2 were provided
/Users/mjackson/Workspace1/complex/src/complex/Filter/Output.hpp:113:23: note: candidate constructor (the implicit default constructor) not viable: requires 0 arguments, but 2 were provided
1 error generated.
```

Reply:
Ah apparently `emplace_back` doesn't work with aggregate initialization. Explicitly defining a constructor would allow it to work. If we don't want to add a constructor, it should be `modifiedActions.push_back(DataModifiedAction{child, DataModifiedAction::ModifiedType::DataArrayModified});` since the advantage of `emplace_back` is constructing in place.